Dockerfile

```diff
@@ -9,7 +9,7 @@ RUN apk add --no-cache build-base
 
 # Copy project deps first since they don't change often
 COPY go.mod go.sum ./
-RUN go mod download
+RUN --mount=type=cache,target=/go/pkg go mod download
```

Comment:
1. nice! learned something new.
2. wonder if we can leverage the go dependency cache (from my previous comment) for the docker build too.